rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java

```diff
@@ -182,23 +183,23 @@ void sample(RrdDb dataBase, long timeInMs) throws IOException {
                 }
             } catch (NumberFormatException ex) {
                 // do not save non-numeric values
-                if (logger.isTraceEnabled()) {
+                String exceptionAsString = StackTraceUtil.getStackTrace(ex).replaceAll("For input string: \".*\"", "");
+                exceptionAsString = dataSource + " / " + fullName + exceptionAsString;
+
+                if (!exceptionsThrown.contains(exceptionAsString)) {
+                    exceptionsThrown.add(exceptionAsString);
+                    logger.warn("Non numeric value for " + dataSource + " / " + fullName + ": " + ex.getMessage());
+                } else {
                     logger.trace("Non numeric value for " + dataSource + " / " + fullName + ": " + ex.getMessage());
                 }
-            } catch (InstanceNotFoundException | AttributeNotFoundException e) {
-                if (logger.isTraceEnabled()) {
-                    logger.trace("Cannot read attribute " + attrName + " for object " + objectName + ": " +
-                                 e.getMessage());
-                }
-            } catch (UnsupportedOperationException e) {
-                if (logger.isTraceEnabled()) {
+            } catch (Exception e) {
+                String exceptionAsString = StackTraceUtil.getStackTrace(e);
+                if (!exceptionsThrown.contains(exceptionAsString)) {
+                    exceptionsThrown.add(exceptionAsString);
                     logger.warn("Error while reading attribute " + attrName + " for object " + objectName + ": ", e);
```

Comment:
We remove isTraceEnabled here, because the real check is done by  if (!exceptionsThrown.contains(exceptionAsString)).

The idea is to print exceptions only once. Adding an extra isWarnEnabled() will not bring anything in terms of performance.